### PR TITLE
Search criteria

### LINKED ISSUE #108: Search criteria
- [x] Filter: unit
- [x] Filter: resource name
- [ ] Filter: Start and end
- [x] Filter: People
- [x] Filter: Equipment
- [x] Filter: Type
- [ ] Show/hide detailed search criteria
- [ ] Styles

Keep the search button for now. It will be removed in another issue.

Backend should support these params:
- unit (id)
- is_favorite (boolean)
- type (list of ids)
- equipment (list of ids)
- people (number)
- start and end (datetime)

### PR DESCRIPTION
**Pull request by evehmane**
Closes #108 

Date time range filter ("Vapaana aikavälille") will be implemented in another branch.

### CODE REVIEW

File: app/pages/search/search-controls/SearchControls.js [hunk 1]
```diff
@@ -1,30 +1,59 @@
...
 import DatePicker from 'shared/date-picker';
+import LabelSelect from 'shared/form-fields/label-select/LabelSelect';
+
+function getUnitOption(id, label) {
+  return <option key={id} value={id}>{label}</option>;
+}
+
+function renderUnitOptions(units) {
+  const defaultOption = getUnitOption('', 'Kaikki kiinteistöt');
+  const options = Object.keys(units).map((id) => {
```
Comment (FelixLoether) [inline app/pages/search/search-controls/SearchControls.js]: - [ ] Is the order of the keys returned by Object.keys defined? I think it's arbitrary order, so it'd be good to sort these for example alphabetically. Same with the other uses of Object.keys in this file.

File: app/shared/form-fields/label-select/LabelSelect.spec.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        expect(options.at(0).prop('bsStyle')).to.equal(defaults.selectedStyle);
+        expect(options.at(1).prop('bsStyle')).to.equal('default');
+        expect(options.at(2).prop('bsStyle')).to.equal(defaults.selectedStyle);
+      });
+    });
+  });
+
+  describe('handleSelect', () => {
+    afterEach(() => {
+      simple.restore();
```
Comment (FelixLoether) [inline app/shared/form-fields/label-select/LabelSelect.spec.js]: - [ ] This afterEach doesn't seem necessary.

Comment (evehmane) [inline app/shared/form-fields/label-select/LabelSelect.spec.js]: Removed (and squashed).

File: app/pages/search/search-controls/SearchControls.js (hunk 1 above)
Comment (evehmane) [inline app/pages/search/search-controls/SearchControls.js]: Fixed.

**Pull request merged**